### PR TITLE
CLOUDP-177769: remove duplicated tests from atlas cli

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes
remove duplicated tests from atlas cli

_Jira ticket:_ CLOUDP-177769

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

## Further comments

After merging this PR the following checks should be made mandatory in Github
- Code Health / libraryOwners
- Code Health / lint
- Code Health / tests (ubuntu-latest)
- Code Health / shellcheck

maybe these two should be kept optional? perhaps not run on every PR only on master?
Code Health / tests (windows-latest)
Code Health / tests (macos-latest)

### CODE REVIEW

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Review by andreaangiolillo — COMMENTED** [review]
Can we make all of these workflows part of a single code_health workflow? similar to what we have with CFN? https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/.github/workflows/code-health.yaml

**Review by andreaangiolillo — DISMISSED** [review]
LGTM. Thanks for all the changes here

**Review by andreaangiolillo — APPROVED** [review]

**Comment by fmenezes** [comment]
What do we need the xunit reports for? Do we use these reports in any way?

**Comment by fmenezes** [comment]
Coverage is being reported in the test output. Where do we report coverage today? In evergreen?

**Comment by fmenezes** [comment]
I think for breaking and foliage I personally think GitHub history is good enough to keep track of which change broke what test.

If that is not sufficient we can just revert this PR later on.

**Comment by gssbzn** [comment]
>What do we need the xunit reports for? Do we use these reports in any way?

Yes, it's what gives us this nice output

Screenshot 2023-05-18 at 20 50 02

I assume right now if the test fails we'll have to grep the logs, this is a bad step backwards personally 

> Where do we report coverage today? In evergreen?

Yes, see the attached [report](https://mongodb-mongocli-build.s3.amazonaws.com/mongodb-atlas-cli-master/coverage/internal/mongodb_atlas_cli_master_code_health_unit_test_12114345f0a75f4116d79240748758e358e02fc9_23_05_18_16_02_13.html#file0), I find them useful everynow and then but not a deal breaker

having to grep logs for test failues I do think is a step back and also why I'd like the split of the tests so I don't have to grep 3 different types of tests to find the failure

**Comment by fmenezes** [comment]
@gssbzn for the coverage reports (example artifacts section of https://github.com/mongodb/mongodb-atlas-cli/actions/runs/5024698073) and also the summary of the tests (based on junit) we have github actions summary (example in https://github.com/mongodb/mongodb-atlas-cli/actions/runs/5024698073/attempts/1#summary-13609959296)

When it comes to grepping logs you're absolutely correct we don't have a web UI to find the logs, I've used `gotestsum` to have a similar report to the one we were used to. Combined with the test summary it will tell what test went wrong hence you can find your logs based on test name.

note: I _personally_ never use the "logs per test" view always grepped in the browser for evergreen so no change to me.

note 2: I understand this is a step back to some of us I'm thinking now is to merge the Github Actions improvements without the removal of Evergreen, let it be for a while. Revisit it at a later stage, what do you think?

**Review by gssbzn — COMMENTED** [review]
In general I'm ok with deleting the evg side as long as we understand the pros and cons, and why I flagged some of them, so there are no surprises

File: .github/workflows/code-health.yml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      - run: go install gotest.tools/gotestsum@latest
+      - run: make unit-test
+      - run: go tool cover -html=coverage.out -o coverage.html
+        if: matrix.os == 'ubuntu-latest'
+      - name: Upload coverage
+        uses: actions/upload-artifact@v2
+        if: matrix.os == 'ubuntu-latest'
+        with:
+          name: coverage
+          path: coverage.html
```
[the remark this answers is not in the document]
Comment (fmenezes) [reply .github/workflows/code-health.yml]: removed

**Pull request merged**